Model/MetaDataModel.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Model;
+
+use OxidEsales\Eshop\Core\DatabaseProvider;
+use OxidEsales\Eshop\Core\Field;
+use OxidEsales\Eshop\Core\Registry;
+
+use OxidEsales\EshopCommunity\Core\Model\BaseModel;
+
+/**
+ * Adds meta data functionality to a model.
+ *
+ * @since 1.2.0
+ */
+trait MetaDataModel
```

Comment:
I love the solution, i'm not sure about the name. As for me it is a settings table for are payment methods. But in if you will reuse it in the order table or in other tables its ok.

Reply:
This can be reused for any model. See [above](#discussion_r289730250).